### DISCUSSION TITLE
Conflicting keybindings in Gnome or other environments

### DISCUSSION THREAD

**Discussion by schewara** [ghostty-org/ghostty#3975] [Feature Requests, Ideas]
I was just giving it a test spin and ran into an issue with overlapping key bindings in Gnome and `ghostty`.

While I did not yet check the whole list for more overlaps, I will mainly focus here on the following bindings, which affect me, as the basis for starting the discussion.

```
ctrl  + alt   + up                     goto_split:top
ctrl  + alt   + left                   goto_split:left
ctrl  + alt   + down                   goto_split:bottom
ctrl  + alt   + right                  goto_split:right
```
Are overlapping and causing conflicts with the following Gnome settings.

```
org.gnome.desktop.wm.keybindings switch-to-workspace-left ['<Super>Page_Up', '<Super><Alt>Left', '<Control><Alt>Left']
org.gnome.desktop.wm.keybindings switch-to-workspace-right ['<Super>Page_Down', '<Super><Alt>Right', '<Control><Alt>Right']
org.gnome.desktop.wm.keybindings switch-to-workspace-up ['<Control><Alt>Up']
org.gnome.desktop.wm.keybindings switch-to-workspace-down ['<Control><Alt>Down']
```

> output from `gsettings list-recursively | grep org.gnome.desktop.wm.keybindings|grep -v '\[\]'` as the UI doesn't show all assigned keybindings.

and in the end are not working (at least for me).

While I could of course raise a Github issue for it, I would also think the same problem can possibly pop up in any other Desktop Environment in different variations, I was wondering what the preferred solution of solving issues like this would or should be.

For now I will probably settle for setting custom keybindings, but having these "known-issues" documented somewhere and to know what to do when hitting the same issue, would make it easier for everyone to do the right thing.

**Comment by rhodes-b** [comment]
Are these custom GNOME keybindings. This should not be an issue for the default GNOME environment as that is what Ghostty targets as the most native experience possible for linux

**Reply by schewara** [reply]
They are default.

Even though they are not listed in the [help pages](https://help.gnome.org/users/gnome-help/stable/shell-keyboard-shortcuts.html.en), I was able to find the default settings in the [gsettings-desktop-schemas](https://gitlab.gnome.org/GNOME/gsettings-desktop-schemas/-/blob/master/schemas/org.gnome.desktop.wm.keybindings.gschema.xml.in?ref_type=heads#L242) repo.

**Reply by rhodes-b** [reply]
Im very surprised nobody else has posted this issue

**Reply by bittner** [reply]
I ran into the same problem and can confirm that this appears to be default key binding for GNOME. I run a rather default GNOME setup on NixOS (unstable) and – just like in Ubuntu that I used before NixOS – virtual desktop navigation key bindings are:

- <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>➡️</kbd> = switch to the virtual desktop on the right
- <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>⬅️</kbd> = switch to the virtual desktop on the left

When GNOME still had the vertical layout for virtual desktops the key binding used to be the same, just with <kbd>⬆️</kbd> and <kbd>⬇️</kbd>.

I have set the key bindings to <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>➡️</kbd>/<kbd>⬅️</kbd>/<kbd>⬆️</kbd>/<kbd>⬇️</kbd> for me to work around the issue.

**Comment by rumpelsepp** [comment]
I discovered this issue and I am affected as well. I cannot use the mentioned ghostty keybindings for splits, since they conflict with gnome.

What about changing the ghostty defaults to: ctrl + super + up|down|right|left?

**Reply by schewara** [reply]
For myself, I currently settled for overriding the problematic ones with the default [Tilix keybindings](https://github.com/gnunn1/tilix/blob/5289686c66fea93a285eb34f3877e8271178f39f/data/gsettings/com.gexperts.Tilix.gschema.xml#L927) as I was using it before and to lower the switchover pains, as some keybindings already became muscle memory, which need some time to re-train.

**Comment by bravous** [comment]
To be zero configuration ghossty needs to support keybindings that work out of the box in gnome. The following is a list of bindings are already used by default gnome:

```
ctrl  + alt   + up          goto_split:up
ctrl  + alt   + down        goto_split:down
ctrl  + alt   + right       goto_split:right
ctrl  + alt   + left        goto_split:left
```

Is it an idea to change this to `ctrl + super + up/down/left/right`
